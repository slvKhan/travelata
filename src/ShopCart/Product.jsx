/* eslint-disable react/prop-types */
import React from 'react';
import { Card, Button } from 'react-bootstrap';

const Product = ({ feats }) => {
  const {
    id, productName, price, amount,
  } = feats;

  return (
    <Card id={id}>
      <Card.Body className="row">
        <span className="amount col-3">
          <span>Осталось</span>
          <span> </span>
          <span>{amount}</span>
        </span>

        <span className="product-name col-3">
          {productName}
        </span>

        <span className="price col-3">
          <span>по</span>
          <span> </span>
          <span>{price}</span>
          <span> </span>
          <span>рублей</span>
        </span>

        <Button variant="primary" className="col-3">Добавить в корзину</Button>
      </Card.Body>
    </Card>
  );
};

export default Product;
