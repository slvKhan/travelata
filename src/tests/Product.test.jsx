/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Product from '../ShopCart/Product';

test('Product render_1', () => {
  const productName = 'Телефон';
  const id = 1;
  const price = 10;
  const amount = 1;

  const addProduct = () => null;
  const btnDisabled = false;
  const product = {
    id,
    productName,
    price,
    amount,
  };

  const component = renderer.create(<Product
    key={product.id}
    product={product}
    btnDisabled={btnDisabled}
    addProduct={addProduct}
  />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Product render_2', () => {
  const productName = 'text 2';
  const id = 5;
  const price = 123;
  const amount = 73;

  const addProduct = () => null;
  const btnDisabled = true;
  const product = {
    id,
    productName,
    price,
    amount,
  };

  const component = renderer.create(<Product
    key={product.id}
    product={product}
    btnDisabled={btnDisabled}
    addProduct={addProduct}
  />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
