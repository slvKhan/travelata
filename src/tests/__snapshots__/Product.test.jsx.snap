// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product render_1 1`] = `
<div
  className="mb-2 card"
  id={1}
>
  <div
    className="row card-body"
  >
    <span
      className="amount col-3"
    >
      <span>
        Осталось
      </span>
      <span>
         1 
      </span>
    </span>
    <span
      className="product-name col-3"
    >
      Телефон
    </span>
    <span
      className="price col-3"
    >
      <span>
        по
      </span>
      <span>
         10 
      </span>
      <span>
        рублей
      </span>
    </span>
    <button
      className="col-3 btn btn-primary"
      disabled={false}
      onClick={null}
      type="button"
    >
      Добавить в корзину
    </button>
  </div>
</div>
`;

exports[`Product render_2 1`] = `
<div
  className="mb-2 card"
  id={5}
>
  <div
    className="row card-body"
  >
    <span
      className="amount col-3"
    >
      <span>
        Осталось
      </span>
      <span>
         73 
      </span>
    </span>
    <span
      className="product-name col-3"
    >
      text 2
    </span>
    <span
      className="price col-3"
    >
      <span>
        по
      </span>
      <span>
         123 
      </span>
      <span>
        рублей
      </span>
    </span>
    <button
      className="col-3 disabled btn btn-primary"
      disabled={true}
      onClick={null}
      type="button"
    >
      Добавить в корзину
    </button>
  </div>
</div>
`;
